near-sdk/src/environment/env.rs

```diff
@@ -1136,21 +1140,25 @@ pub fn promise_batch_then(promise_index: PromiseIndex, account_id: &AccountId) -
 }
 
 /// Set the account id that will receive the refund if the promise panics.
-/// Uses low-level [`crate::env::promise_set_refund_to`]
+/// Uses low-level [`crate::sys::promise_set_refund_to`]
+///
 /// # Examples
 /// ```
+/// # #[cfg(feature = "deterministic-account-ids")]
+/// # let _ = {
 /// use near_sdk::env::{promise_set_refund_to, promise_create};
-/// use near_sdk::AccountId;
+/// use near_sdk::{AccountId, Gas, NearToken};
 /// use std::str::FromStr;
 ///
 /// let promise = promise_create(
 ///     "account.near".parse().unwrap(),
 ///     "method",
 ///     [],
 ///     NearToken::from_millinear(1),
-///     NearGas::from_tgas(1),
+///     Gas::from_tgas(1),
 /// );
-/// promise_set_refund_to(promise, "refund.near".parse().unwrap());
+/// promise_set_refund_to(promise, &"refund.near".parse().unwrap());
+/// # };
 /// ```
 #[cfg(feature = "deterministic-account-ids")]
 pub fn promise_set_refund_to(promise_index: PromiseIndex, account_id: &AccountId) {
```

Comment:
```suggestion
pub fn promise_set_refund_to(promise_index: PromiseIndex, account_id: impl AsRef<AccountIdRef>) {
```

Reply:
I think you have already prepared the fix for this in #1458, right?